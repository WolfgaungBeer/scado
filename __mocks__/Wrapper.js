import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
