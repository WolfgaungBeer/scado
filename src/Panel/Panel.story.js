import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Panel from './';


storiesOf('Panel', module)
    .add('default', () => (
        <Panel>test</Panel>
    ))
    .add('with props', () => (
        <Panel
            backgroundColor={text('backgroundColor', 'white')}
            borderColor={text('borderColor', 'lightGray')}
            borderRadius={text('borderRadius', 's')}
            shadow={text('shadow', 's')}
        >
            test
        </Panel>
    ));
