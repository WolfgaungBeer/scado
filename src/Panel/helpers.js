export const getShadow = () => (props) => {
    if (!props.shadow) return '';
    const shadowValue = props.theme.shadow[props.shadow];
    return `${shadowValue} ${shadowValue} 5px 0px rgba(0,0,0,0.75);`;
};
