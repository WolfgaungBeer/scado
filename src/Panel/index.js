import styled from 'styled-components';
import { string, node } from 'prop-types';
import { getShadow } from './helpers';

const propTypes = {
    backgroundColor: string,
    borderColor: string,
    borderRadius: string,
    shadow: string,
    className: string,
    children: node,
};

const defaultProps = {
    backgroundColor: undefined,
    borderColor: undefined,
    borderRadius: undefined,
    shadow: undefined,
    className: undefined,
    children: undefined,
};

const Panel = styled.div`
    width: 100%;
    height: 100%;
    background-color: ${props => props.theme.color[props.backgroundColor]};
    border-style: ${props => props.borderColor && 'solid'};
    border-width: 1px;
    border-color: ${props => props.theme.color[props.borderColor]};
    border-radius: ${props => props.theme.scale[props.borderRadius]};
    box-sizing: border-box;
    box-shadow: ${getShadow()};
`;

Panel.propTypes = propTypes;
Panel.defaultProps = defaultProps;

export default Panel;
