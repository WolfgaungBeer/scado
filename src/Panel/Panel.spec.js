import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import Panel from './';

it('renders the Icon Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Panel>check_circle</Panel>
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Panel backgroundColor="error" borderColor="success">check_circle</Panel>
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <Panel backgroundColor="white" borderRadius="xs" shadow="s" >check_circle</Panel>
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();
});
