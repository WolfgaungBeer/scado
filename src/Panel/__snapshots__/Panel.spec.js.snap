// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Icon Component correctly 1`] = `
<div
  className="sc-eNQAEJ kUfWEU"
>
  check_circle
</div>
`;

exports[`renders the Icon Component correctly 2`] = `
<div
  className="sc-eNQAEJ fDzKqi"
>
  check_circle
</div>
`;

exports[`renders the Icon Component correctly 3`] = `
<div
  className="sc-eNQAEJ dxwLkT"
>
  check_circle
</div>
`;
